### PR TITLE
[WAGON-570] Use RedirectStrategy from HttpClient rather than our cust…

### PR DESCRIPTION
**Pull request by michael-o**
…om code

@ok2c Please have a look here.

### CODE REVIEW

File: wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/WagonRedirectStrategy.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        {
+            return new HttpHead( uri );
+        }
+        else if ( method.equalsIgnoreCase( HttpGet.METHOD_NAME ) )
+        {
+            return new HttpGet( uri );
+        }
+        else
+        {
+            return RequestBuilder.copy( request ).setUri( uri ).build();
```
Comment (ok2c) [inline wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/WagonRedirectStrategy.java]: @michael-o I suppose there is no need to treat `HEAD` and `GET` as special cases. `RequestBuilder.copy( request ).setUri( uri ).build();` should be OK for all methods. 
Looks good to me.

Comment (michael-o) [reply wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/WagonRedirectStrategy.java]: That's actually from here:  https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/client/DefaultRedirectStrategy.java#L222-L231

Comment (ok2c) [reply wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/WagonRedirectStrategy.java]: @michael-o I know. There however it actually makes sense. By default HttpClient 4.5 makes a full copy of the redirected request only in response to status 307.

**Comment by jira-importer** [comment]
Resolve #636

**Pull request closed without merging**